arroyo-sql/src/operators.rs

```diff
@@ -159,6 +160,35 @@ impl CodeGenerator<ValuePointerContext, StructDef, syn::Expr> for UnnestProjecti
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct AsyncUdfProjection {
+    pub column: Column,
+    pub expression: Expression,
+    pub async_udf: RustUdfExpression,
+}
+
+impl AsyncUdfProjection {
+    pub fn output_struct(&self) -> StructDef {
+        self.expression_type(&ValuePointerContext::new())
+    }
+}
+
+impl CodeGenerator<ValuePointerContext, StructDef, syn::Expr> for AsyncUdfProjection {
+    fn generate(&self, _input_context: &ValuePointerContext) -> syn::Expr {
+        unreachable!()
```

Comment:
This seems like not the approach. The idea behind the `CodeGenerator<Context, OutputValue, OutputType: ToTokens>` trait is that you specify how to produce some syn type ToTokens, given a Context and an OutputValue. I think you want the code after `  RecordTransform::AsyncUdfProjection(a) => {` to be specified in the generate() method of a CodeGenerator implementation.